### PR TITLE
Navigation: Some navigation menu items left out due to duplicate IDs

### LINKED ISSUE #5626: Some navigation items missing due to duplicate IDs
**Describe the bug**

@elizaan36 pointed out that the "Products" item is missing from the "Settings" category in [another PR](https://github.com/woocommerce/woocommerce-admin/issues/5572#issuecomment-726889677). It looks like this is actually due to conflicting IDs, an issue that could also easily cause issues elsewhere. 

We need to rethink a bit how these IDs are generated to prevent these kinds of conflicts. 

**To Reproduce**

1. Ensure the new navigation is enabled
2. Navigate to WooCommerce -> Settings
3. Observe that the "Products" item is no longer present

**Expected behavior**
All expected items should be present.

### PR DESCRIPTION
**Pull request by joelclimbsthings**
Fixes #5626

This issue was noticed due to the "Products" menu item disappearing from the "Settings" category. On closer inspection, it looks like this stems from that use of fairly generic IDs (ie "products") resulting in conflicts that resulted in some items not being added as they should. For instance, in this case, a menu item had already been registered with the ID of "products," which resulted in the Settings page with the same ID being dismissed. 

To remedy this I've modified the code to identify each item in the context of their `parent` as well, so the resulting ID now follows this format: `parent-name/item-id`. 

This should allow similar items to be added without issue as long as they exist in a different level/category of the navigation hierarchy. 

This PR also includes updating how current slot-fill items are registered to support the new ID structure, as well as updating the documentation to reflect this change. 

Last, we're now printing a warning to the `error_log()` when a a duplicate item attempts registration. I'm not entirely confident of the suggested format for this since I could only find what example of it's use in the project.

### Screenshots

Settings -> Products item now visible:
image

### Detailed test instructions:

- Checkout branch
- Ensure navigation feature is enabled
- Navigation to WooCommerce -> Settings
- Ensure that the "Products" menu item is now visible in this category, as in the screenshot
- Click back and forth from a few react-based pages (Customer/Home/Analytics reports) and ensure there is still no page refresh.

### CODE REVIEW

File: src/Features/Navigation/Menu.php [hunk 1]
```diff
@@ -231,6 +233,19 @@ private static function add_item( $args ) {
...
+		if ( isset( self::$menu_items[ $menu_item['id'] ] ) ) {
+			error_log(  // phpcs:ignore
+				sprintf(
+					/* translators: 1: Duplicate menu item path. */
+					esc_html__( 'You have attempted to register a duplicate item with WooCommerce Navigation: %1$s', 'woocommerce-admin' ),
+					'`' . $menu_item['id'] . '`'
+				)
+			);
+			return;
+		}
```
Comment (psealock) [inline src/Features/Navigation/Menu.php:247]: Nice addition here Joel

File: client/layout/controller.js [hunk 2]
```diff
@@ -60,6 +60,7 @@ export const getPages = () => {
 		],
 		wpOpenMenu: 'toplevel_page_woocommerce',
 		id: 'home',
+		navParent: 'woocommerce',
```
Comment (psealock) [inline client/layout/controller.js:63]: I introduced the `id` parameter just for the purposes of the Nav so I'm confident its not used anywhere else. Given that, do you think it may be worthwhile to just make the id's themselves more unique, ie `id: 'woocommerce/home'`?

**Review by psealock — COMMENTED** [review]
Changes look great @joelclimbsthings, I'm just wondering if we can simplify the `id`.

**Comment by psealock** [comment]
Speaking of id's, I just realized one thing. I introduced the id for use of identification of screens but didn't really like that now any plugin extending wc-admin now had to also add the same id on the client side which would change the API. 

As an example, [see WC Pay adding a Deposits screen in PHP](https://github.com/Automattic/woocommerce-payments/blob/11c4f56a605fc947ae5dd3b726a7ca9a1965daec/includes/admin/class-wc-payments-admin.php#L73-L80). An `id` is present on the PHP side, but in order to match id's to get SlotFill working I addded the `id` parameter on the client as well. In this PR you're manipulating wc-admin's implementation of this id. On the client side, [here is WC Pay registering the same Deposits page](https://github.com/Automattic/woocommerce-payments/blob/c7f9d58e2d20fc5fb10d3fa5d205f698bcf3b8c6/client/index.js#L35-L40). In order for slotFill to work they will need to supply the same id on the client as they did the PHP side. I didn't like that requirement because I wanted it to work "out of the box", so I created https://github.com/woocommerce/woocommerce-admin/issues/5458 to track a solution.

Sooooo, the short of the long story is: Can we use `path` parameter instead as a unique identifier? That parameter is already available on PHP and JS so no update would be required on wc-admin extensions? If that isn't unique enough, can we use something else? Maybe serialize one of the other parameters?

If so, this PR would solve https://github.com/woocommerce/woocommerce-admin/issues/5458 as well. I hope this makes sense and my rambling isn't too hard to follow.

File: client/layout/controller.js (hunk 2 above)
Comment (joshuatf) [reply client/layout/controller.js:63]: +1

File: src/Features/Navigation/Menu.php [hunk 3]
```diff
@@ -231,6 +233,19 @@ private static function add_item( $args ) {
...
 		}
 
+		$menu_item['id'] = $menu_item['parent'] . '/' . $menu_item['id'];
+
+		if ( isset( self::$menu_items[ $menu_item['id'] ] ) ) {
+			error_log(  // phpcs:ignore
+				sprintf(
+					/* translators: 1: Duplicate menu item path. */
+					esc_html__( 'You have attempted to register a duplicate item with WooCommerce Navigation: %1$s', 'woocommerce-admin' ),
+					'`' . $menu_item['id'] . '`'
```
Comment (joshuatf) [inline src/Features/Navigation/Menu.php:243]: ❤️

File: client/layout/controller.js (hunk 2 above)
Comment (joelclimbsthings) [reply client/layout/controller.js:63]: Definitely :+1:

**Comment by joelclimbsthings** [comment]
Thanks for the thoughts @psealock , I really like the idea of automating the client-side registration. It sounds like the intention would only be to do so for pages already registering through `wc-admin`, right? It would be easy enough to use the `path` as the unique ID, with the only trick being how we expose this to the client-side. 

We _do_ already have `getPages()` in `controller.js`, but these paths are currently set manually for the most part. I know we have tools such as `wp_enqueue_script()` to surface data from the server-side in the client side, but I haven't used them as of yet. Do you have any thoughts around how we tie these together, and if we want to do so in the context of this PR?

**Comment by joshuatf** [comment]
>It sounds like the intention would only be to do so for pages already registering through wc-admin, right?

I'm not sure if @psealock's intention was to replace `id` with `path` just for WCA items or to make this the new ID for all nav items.  The latter is an interesting idea, but we'd need to determine if we're okay with not allowing duplicate paths in the same category.  Some plugins do take advantage of this cross-category, but I think it may be an edge case where a plugin would want to register the exact same path twice in a single category.

>I know we have tools such as wp_enqueue_script() to surface data from the server-side in the client side, but I haven't used them as of yet.

`wp_localize_script` is the WordPress way to pass server vars to client scripts.

>Do you have any thoughts around how we tie these together, and if we want to do so in the context of this PR?

I think @psealock's method avoids passing server to client info and relies on the path being the identifier.

**Comment by joelclimbsthings** [comment]
Closing in favor of https://github.com/woocommerce/woocommerce-admin/pull/5665#issuecomment-732470201
